Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -146,6 +154,34 @@ namespace AZ
             }
         }
         
+        void MaterialAsset::RealignPropertyValuesAndNames()
+        {
+            const MaterialPropertiesLayout* propertyLayout = GetMaterialPropertiesLayout();
+            AZStd::vector<MaterialPropertyValue> alignedPropertyValues(m_materialTypeAsset->GetDefaultPropertyValues().begin(), m_materialTypeAsset->GetDefaultPropertyValues().end());
+            for (size_t i = 0; i < m_propertyNames.size(); ++i)
+            {
+                const MaterialPropertyIndex propertyIndex = propertyLayout->FindPropertyIndex(m_propertyNames[i]);
+                if (propertyIndex.IsValid())
+                {
+                    alignedPropertyValues[propertyIndex.GetIndex()] = m_propertyValues[i];
+                }
+                else
+                {
+                    AZ_Warning(s_debugTraceName, false, "Material property name \"%s\" is not found in the material properties layout and will not be used.", m_propertyNames[i].GetCStr());
+                }
```

Comment:
Does this part supports property name change? 
There should be handling case by unique Id to realign to the new name, hence have backwards compatibility.